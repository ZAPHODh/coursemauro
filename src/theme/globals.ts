'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html {
        font-family: ${montserrat.style.fontFamily};
        max-width: 100vw;
    }
    ,
    main {
        max-width: 100vw;
    }
    body {
        padding: 0;
        margin: 0;
        overflow: hidden;
    }
    ,
    input {
        font-family: inherit;
    }
    ,
    a {
        color: inherit;
    }
    ,
    * {
        box-sizing: border-box;
    }
    ,
    button {
        font-family: inherit;
    }
`}`
