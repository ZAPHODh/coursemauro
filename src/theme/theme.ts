import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
}
