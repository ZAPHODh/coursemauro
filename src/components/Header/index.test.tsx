import { render } from '@testing-library/react'
import { Header } from '.'
import { ThemeClient } from '../../theme/ThemeClient'

describe('Header', () => {
    it('should render the header', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Header course="test" description="test" />
            </ThemeClient>
        )

        const heading = getByRole('banner')

        expect(heading).toBeInTheDocument()
    })
    it('should render the img properly', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Header course="test" description="test" />
            </ThemeClient>
        )

        const img = getByRole('img')

        expect(img).toBeInTheDocument()
    })
    it('should take a headers snapshot', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Header course="test" description="test" />
            </ThemeClient>
        )

        const heading = getByRole('banner')

        expect(heading).toMatchSnapshot()
    })
})
