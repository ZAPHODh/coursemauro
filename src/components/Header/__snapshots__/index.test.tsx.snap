// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should take a headers snapshot 1`] = `
<header
  class="sc-925b7b98-0 fvnARU"
>
  <div>
    <h1
      class="sc-e26ab1cd-0 idkQwD"
    >
      test
    </h1>
    <p
      class="sc-a2a9610f-0 dePwer"
      maxwidth="440px"
      padding="5px"
    >
      test
    </p>
    <button
      class="sc-78528855-0 fAPeaZ"
      font-size="inherit"
      width="90%"
    >
      QUERO DOMINAR AS TÃ‰CNICAS
    </button>
  </div>
  <img
    alt="foto-mauro"
    data-nimg="1"
    decoding="async"
    height="537"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
    style="color: transparent;"
    width="376"
  />
</header>
`;
